test/IECoreScene/ShaderNetworkTest.py

```diff
@@ -0,0 +1,504 @@
+		self.assertTrue( IECoreScene.ShaderNetwork.Parameter( "x", "" ) )
+		self.assertTrue( IECoreScene.ShaderNetwork.Parameter( "x", "y" ) )
+		self.assertTrue( IECoreScene.ShaderNetwork.Parameter( "", "y" ) )
+
+	def testConnection( self ) :
+
+		c = IECoreScene.ShaderNetwork.Connection()
+		self.assertEqual( c.source.shader, "" )
+		self.assertEqual( c.source.name, "" )
+		self.assertEqual( c.destination.shader, "" )
+		self.assertEqual( c.destination.name, "" )
+
+		p1 = IECoreScene.ShaderNetwork.Parameter( "x", "out" )
+		p2 = IECoreScene.ShaderNetwork.Parameter( "y", "in" )
+
+		c = IECoreScene.ShaderNetwork.Connection( p1, p2 )
+		self.assertEqual( c.source, p1 )
+		self.assertEqual( c.destination, p2 )
+
+		c.source.name = "out.r"
+		self.assertEqual( c.source.name, "out.r" )
+
+		self.assertEqual(
+			repr( c ),
+			'Connection( Parameter( "x", "out.r" ), Parameter( "y", "in" ) )'
+		)
+
+	def testConstructor( self ) :
+
+		shaders = {
+			"out" : IECoreScene.Shader( "flat", "surface" ),
+			"texture" : IECoreScene.Shader( "noise", "shader" ),
+		}
+
+		connections = [
+			IECoreScene.ShaderNetwork.Connection(
+				IECoreScene.ShaderNetwork.Parameter(
+					"texture", "outColor"
+				),
+				IECoreScene.ShaderNetwork.Parameter(
+					"out", "Cs"
+				)
+			)
+		]
+
+		n = IECoreScene.ShaderNetwork(
+			shaders = shaders,
+			connections = connections,
+			output = IECoreScene.ShaderNetwork.Parameter( "out" )
+		)
+		self.assertEqual( n.shaders(), shaders )
+		self.assertEqual( n.inputConnections( "out" ), connections )
+		self.assertEqual( n.outputConnections( "texture" ), connections )
+		self.assertEqual( n.getOutput(), IECoreScene.ShaderNetwork.Parameter( "out" ) )
+
+	def testShaderAccessors( self ) :
+
+		n = IECoreScene.ShaderNetwork()
+
+		s1 = IECoreScene.Shader()
+		s2 = IECoreScene.Shader()
+
+		n.addShader( "s1", s1 )
+		self.assertFalse( n.getShader( "s1" ).isSame( s1 ) )
+		self.assertEqual( n.getShader( "s1" ), s1 )
+		self.assertEqual( n.shaders(), { "s1" : s1 } )
+
+		n.addShader( "s2", s2 )
+		self.assertEqual( n.shaders(), { "s1" : s1, "s2" : s2 } )
+		self.assertFalse( n.getShader( "s2" ).isSame( s2 ) )
+		self.assertEqual( n.getShader( "s2" ), s2 )
+
+		n.removeShader( "s1" )
+		with self.assertRaises( RuntimeError ) :
+			n.removeShader( "s1" )
+		self.assertEqual( n.shaders(), { "s2" : s2 } )
+
+		self.assertEqual( n.getShader( "s1" ), None )
+
+	def testOutput( self ) :
+
+		n = IECoreScene.ShaderNetwork()
+		self.assertEqual( n.getOutput(), n.Parameter() )
+
+		s1 = IECoreScene.Shader( "constant" )
+		s2 = IECoreScene.Shader( "noise" )
+
+		n.addShader( "s1", s1 )
+		self.assertEqual( n.getOutput(), n.Parameter() )
+		self.assertEqual( n.outputShader(), None )
+
+		n.addShader( "s2", s2 )
+		self.assertEqual( n.getOutput(), n.Parameter() )
+		self.assertEqual( n.outputShader(), None )
+
+		n.setOutput( n.Parameter( "s1", "" ) )
+		self.assertEqual( n.getOutput(), n.Parameter( "s1", "" ) )
```

Comment:
I did it deliberately to avoid verbosity. They both refer to the exact same thing, except the latter is actually the even more verbose `IECoreScene.ShaderNetwork.Parameter()`. If you think it's misleading I can replace them all with the long form, or I can use the implicit tuple->Parameter conversion I added later as a convenience.